lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -530,12 +526,15 @@ function dosomething_global_redirect_node_anon() {
  * @param int $redirect_status
  */
 function dosomething_global_redirect_node_language($node, $language_code, $redirect_status = 302) {
-
   $languages = language_list();
 
   // Can only redirect to an published, translated version.
-  if (!empty($node->translations->data[$language_code]) && (bool)$node->translations->data[$language_code]['status']) {
```

Comment:
couldn't we also return a drupal not found from the bottom of this if statement? it was previously checking status